services/venice-controller/src/main/java/com/linkedin/venice/controller/VeniceHelixAdmin.java

```diff
@@ -5432,10 +5436,31 @@ private Pair<ExecutionStatus, String> getDaVinciPushStatusAndDetails(
         incompletePartition.add(partitionId);
       }
     }
+
+    if (totalInstanceCount == 0) {
+      return new Pair<>(completeStatus, "No DaVinci instance found yet.");
+    }
+
+    // Report error if too many davinci instances are not alive for over 5 mins
+    if (liveInstanceCount < 0.5 * totalInstanceCount) {
+      Long lastUpdateTime = storeVersionToDVCDeadInstanceTimeMap.get(storeName + version.getNumber());
+      if (lastUpdateTime != null) {
+        if (lastUpdateTime + DAVINCI_DEAD_INSTANCE_REPORT_ERROR_WAIT < System.currentTimeMillis()) {
+          storeVersionToDVCDeadInstanceTimeMap.remove(storeName + version.getNumber());
+          return new Pair<>(
+              ExecutionStatus.ERROR,
+              " To many dead instances " + (totalInstanceCount - liveInstanceCount) + ", total instances "
+                  + totalInstanceCount);
+        }
+      } else {
+        storeVersionToDVCDeadInstanceTimeMap.put(storeName + version.getNumber(), System.currentTimeMillis());
```

Comment:
This doesn't look right since the function is triggered when AZ poling the job status.
If the AZ job polling interval is bigger than 5 mins, the update interval recorded in `storeVersionToDVCDeadInstanceTimeMap` will be higher than 5 mins.

Also if the `liveInstanceCount` > 0.5 * totalInstanceCount later, shall we reset the tracking map?

Reply:
@sixpluszero is working on another change to refactor the deciding logic of push job completed, where Controller would spawn some async thread to poll push status system store, and I think this idea can be implemented better on top of Jialin's change.

Reply:
I see default AZ job polling period  is 5 mins, so we should definitely wait for more than than, may be 10mins or so. I will check with Jialin on this. Are we trying to converge both AZ job poll status and backend push status in the same code path?